doc/development_process/documentation.rst

```diff
@@ -0,0 +1,87 @@
+
+Code Documentation
+###################
+
+API Documentation
+******************
+
+Well documented APIs enhance the experience for developers and are an essential
+requirement for defining an API's success. Doxygen is a general purpose
+documentation tool that the Zephyr project uses for documenting APIs. It
+generates an on-line documentation browser (in HTML) and provides
+input for other tools used to generate a reference manual from
+documented source files.
+
+Reference to Requirements
+**************************
+
+APIs for the most part document the implementation of requirements or advertised
+features and can be traced back to features. We use the API documentation as the
+main interface to trace implementation back to documented features. This is done
+using custom _doxygen_ tags that reference requirements maintained somewhere
+else in a requirements catalog.
+
+Test Documentation
+*******************
+
+To help understand what each test does and which functionality it tests, we also
+document all test code using the same tools and in the same context, and generate
+documentation for all unit and integration tests maintained in the same
+environment. Tests are documented using references to the APIs or functionality
+they validate by creating a link back to the APIs and by adding a reference to
+the original requirements.
+
+
+Documentation Guidelines
+*************************
+
+Test Code
+**********
+
+Zephyr comes with many test types and methods of testings, the most common test
+method Ztest. The documentation of tests should only be done on the entry test
+functions (usually prefixed with test\_) and those that are called directly by
```

Comment:
"should only be done as doxygen comments in specially designated doxygen groups"?

Reply:
hmm, not the same thing. The idea is to document the entry test functions and not every function available in the test code.